app/controllers/admin/roles_controller.rb

```diff
@@ -0,0 +1,71 @@
+module Admin
+  class RolesController < Admin::BaseController
+    load_and_authorize_resource :conference, find_by: :short_title
+    load_resource :user, only: [ :remove_user, :update ]
+    before_action :set_selection
+    # Show flash message with ajax calls
+    after_filter :prepare_unobtrusive_flash, only: [:add_user, :remove_user]
+
+    def index
+      @roles = Role.where(resource: @conference)
+    end
+
+    def show
+      @users = @role.users
+    end
+
+    def edit
+      @users = @role.users
+    end
+
+    def update
+      role_name = @role.name
+
+      if @role.name == 'organizer' && params[:role][:user_ids].reject(&:empty?).blank?
+        flash[:error] = 'Could not remove user. The conference must have at least one organizer!'
+        redirect_to admin_conference_roles_path(@conference.short_title) && return
+      end
+
+      if params[:role] && @role.update_attributes(role_params)
+        flash[:success] = 'Successfully updated role ' + @role.name
+        redirect_to admin_conference_role_path(@conference.short_title, @role.name)
+      else
+        flash[:error] = 'Could not update role! ' + @role.errors.full_messages.to_sentence
+        @role.name = role_name
+        render :edit
+      end
+    end
+
+    def toggle_user
+      if params[:user] && (user_email = params[:user][:email]) && user_email.present?
```

Comment:
Please move 
``
user_email = params[:user][:email]
``
out of the conditional! This is really hard to understand ...